### DISCUSSION TITLE
Domain-driven Laravel preset

### DISCUSSION THREAD

**Discussion by kylemilloy** [pestphp/pest#1334] [Ideas]
It'd be a really nice feature if I could pass a domain to `arch()->preset()->laravel()` and have it add the exceptions for that domain instead of defaulting to `App/*` for everything. For example, my structure looks like:

```
- app
  - Auth
    - Models
    - Http
      - Controllers
    - Tests
  - Dashboard
    - Models
    - Http
      - Controllers
    - Tests
```

And if I had a usage where I could add an ArchTest.php to the base of all my test suites that had:

```php
pest()->preset()->laravel('App/Auth');
```

And this would then see the exceptions in Laravel marked as:

```php
$this->expectations[] = expect('App\Auth\Traits')
    ->toBeTraits();

// etc.
```

I can see that internally the abstract has a concept of namespaces (static::$baseNamespaces) but these are hard-coded in the Laravel preset to `App/*` and look to be unused. Willing to write a PR for this but not sure if the intent would be to use baseNamespace or just make this single preset more flexible.

**Comment by pelmered** [comment]
Great idea, but there are so many different ways to structure your project with a DDD approach. It wouldn't make much sense to just support one structure, but as long as it is flexible and easy to adapt to your structure it would be a great feature. 

For example in most of my projects I use a structure similar to this:

```
├── src
│   ├── App  (All app code, like controllers, middleware, service providers), 
│   ├── Domains (All domain logic)
│   ├── Integrations (Third-party integrations)
│   └── Support (Helper classes, global base classes etc.)
```
